subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTasksIntegrationTest.groovy

```diff
@@ -108,4 +112,70 @@ class FailingIncrementalTasksIntegrationTest extends AbstractIntegrationSpec {
         "remove"      | false       | "with removed outputs is fully rebuilt"
         "none"        | true        | "with unmodified outputs is executed as incremental"
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/9320")
+    def "incremental task with NAME_ONLY input (matching file names and content) detects changed input"() {
+        def taskName = "incrementalTask"
+        def inputs = ('a'..'z').collect { file("${it}/input.txt").createFile() }
+        def modifiedInput = inputs.toList().sort { random() }.first()
```

Comment:
Sorting with `random()` can lead to problems, can't it? I am pretty sure the random function does not adhere to the contract for a comparator, so sorting with it can lead to an exception, right?

How about you select a random element of the list. Sorting the list does not really help.

Or even better, let's not do anything randomly and add a `where` clause which checks the first, last and some element in the middle.

Reply:
Using randomness can also introduce flakiness to the tests that is difficult to debug. Let's use specific values in a `where:` as @wolfs suggested.

Reply:
Those flaky values would find real bugs (i.e. property based test), but I agree that it could be hard to trace, so I'll go deterministic.